app/controllers/providers/about_the_financial_assessments_controller.rb

```diff
@@ -34,5 +35,11 @@ def start_after_means_complete_path
         legal_aid_application:,
       )
     end
+
+    def delete_previously_scheduled_mails
+      ScheduledMailing.where(legal_aid_application_id: legal_aid_application.id, status: "waiting").each do |scheduled|
+        scheduled.delete if scheduled.arguments[1] != legal_aid_application.applicant.email
```

Comment:
should `destroy!` be used to ensure any hooks (current or future) are triggered, and `!` to raise errors?